### PR TITLE
Update to bpftool v7.4

### PR DESCRIPTION
**Pull request by mazdakn**
Update `bpftool` from `v5.3` to `v7.4` to fix this GH issue: https://github.com/projectcalico/calico/issues/8856

The tools is build statically.
Also `armv7` support is dropped.

PR to update the image in Calico-node: https://github.com/projectcalico/calico/pull/8880

### CODE REVIEW

**Comment by hjiawei** [comment]
Maybe build all arches (armv7 can be dropped) from UBI or AlmaLinux so that we don't have glibc abi issues?

**Comment by mazdakn** [comment]
make sense. I'll use one of those.

File: Dockerfile.amd64 [hunk 1]
```diff
@@ -11,34 +11,31 @@
...
 # See the License for the specific language governing permissions and
 # limitations under the License.
-FROM debian:buster-slim as bpftool-build
-ARG KERNEL_REPO=git://git.kernel.org/pub/scm/linux/kernel/git/torvalds/linux.git
-ARG KERNEL_REF=master
+
+FROM debian:bookworm-slim as bpftool-build
+
+ARG SOURCE_REPO=https://github.com/libbpf/bpftool.git
+ARG SOURCE_REF=master
```
Comment (danudey) [inline Dockerfile.amd64]: This should be `main` (there is no `master` branch in `libbpf/bpftool`).

Comment (danudey) [reply Dockerfile.amd64]: (This also applies to all other Dockerfiles)

File: Dockerfile.arm64 [hunk 2]
```diff
@@ -11,40 +11,37 @@
 # WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 # See the License for the specific language governing permissions and
 # limitations under the License.
-ARG QEMU_IMAGE=calico/go-build:v0.20
+
+ARG QEMU_IMAGE=calico/go-build:v0.91
```
Comment (hjiawei) [inline Dockerfile.arm64]: Use calico/qemu-user-static image

**Comment by hjiawei** [comment]
Can you unify Dockerfiles for different arches to be one? You can also use build base from UBI (even though I understand bpftool is static linked).

File: Makefile [hunk 3]
```diff
@@ -74,7 +67,7 @@ image: $(DEFAULTORG)/bpftool
 $(DEFAULTORG)/bpftool: register
 	# Make sure we re-pull the base image to pick up security fixes.
 	# Limit the build to use only one CPU, This helps to work around qemu bugs such as https://bugs.launchpad.net/qemu/+bug/1098729
-	docker build $(DOCKER_BUILD_ARGS) --build-arg KERNEL_REF=$(KERNELREF) --build-arg KERNEL_REPO=$(KERNELREPO) --cpuset-cpus 0 --pull -t $(ARCHIMAGE) -f $(DOCKERFILE) .
+	docker build $(DOCKER_BUILD_ARGS) --platform=linux/$(ARCH) --build-arg SOURCE_REF=$(SOURCEREF) --build-arg SOURCE_REPO=$(SOURCEREPO) --cpuset-cpus 0 --pull -t $(ARCHIMAGE) -f Dockerfile .
```
Comment (hjiawei) [inline Makefile]: Better to use the `docker buildx build` buildkit utils. Also, we can remove the `--cpuset-cpus 0` command line flag. I am not sure why we want to pin to cpu 0.

Comment (mazdakn) [reply Makefile]: The reason for pinning is mentioned in the line above it. Not sure, if it's valid. If you know it's not relevant anymore, I can remove it.

Comment (mazdakn) [reply Makefile]: @hjiawei about using `buildx`, I tried it, and the build failed. For whatever reason, the compiled binary for other arch, were not compiled statically. If there is no important reason to switch, I prefer to not spend more time on it.

Comment (mazdakn) [reply Makefile]: I removed `--cpuset-cpus 0`, and also tried `buildx` one more time. It breaks building other arches. It does not worth spending more time on it.

**Review by hjiawei — APPROVED** [review]

**Review by matthewdupre — APPROVED** [review]

**Pull request merged**